pkg/build/admission/defaults/api/types.go

```diff
@@ -40,6 +40,9 @@ type BuildDefaultsConfig struct {
 
 	// annotations are annotations that will be added to the build pod
 	Annotations map[string]string
+
+	// Resources defines resource requirements to execute the build.
+	Resources *kapi.ResourceRequirements
```

Comment:
We should never use pointers to maps or pointers to slices in our api (because slices and maps in Go are pointers already and we can distinguish between empty and nil, if everything else is done correctly).